### PR TITLE
Improve auto ignore args to allow for macros expanding to a function name

### PR DESCRIPTION
**Pull request by dcristoloveanu**
Improve auto ignore args to allow for macros expanding to a function name

### CODE REVIEW

File: tests/umockautoignoreargs_ut/umockautoignoreargs_ut.c [hunk 1]
```diff
@@ -322,4 +322,106 @@ TEST_FUNCTION(umockautoignoreargs_is_call_argument_ignored_for_2nd_arg_when_firs
...
+
+/* Tests_SRS_UMOCKAUTOIGNOREARGS_01_010: [ umockautoignoreargs_is_call_argument_ignored shall look for the arguments as being the string between the last set of parenthesis in call. ]*/
+TEST_FUNCTION(umockautoignoreargs_is_call_argument_ignored_when_RPAREN_at_end_fails)
+{
+    // arrange
+    int result;
+    int is_ignored;
+
+    // act
+    result = umockautoignoreargs_is_call_argument_ignored("WRAPPER(a)(IGNORED_NUM_ARG(", 1, &is_ignored);
```
Comment (mattdurak) [inline tests/umockautoignoreargs_ut/umockautoignoreargs_ut.c]: >"WRAPPER(a)(IGNORED_NUM_ARG(" [](start = 58, length = 29)

seems to be missing tests with extra RPAREN like:

`("WRAPPER(a)(IGNORED_NUM_ARG))", 1, &is_ignored);`


Or

`("foo(IGNORED_NUM_ARG(", 1, &is_ignored);`

File: tests/umockautoignoreargs_ut/umockautoignoreargs_ut.c [hunk 2]
```diff
@@ -322,4 +322,106 @@ TEST_FUNCTION(umockautoignoreargs_is_call_argument_ignored_for_2nd_arg_when_firs
...
+    // act
+    result = umockautoignoreargs_is_call_argument_ignored("WRAPPER(a)(IGNORED_NUM_ARG)", 1, &is_ignored);
+
+    // assert
+    ASSERT_ARE_EQUAL(int, 0, result);
+    ASSERT_ARE_EQUAL(int, 1, is_ignored);
+}
+
+/* Tests_SRS_UMOCKAUTOIGNOREARGS_01_010: [ umockautoignoreargs_is_call_argument_ignored shall look for the arguments as being the string between the last set of parenthesis in call. ]*/
+TEST_FUNCTION(umockautoignoreargs_is_call_argument_ignored_when_RPAREN_at_end_fails)
```
Comment (mattdurak) [inline tests/umockautoignoreargs_ut/umockautoignoreargs_ut.c]: >RPAREN [](start = 64, length = 6)

LPAREN?

File: tests/umockautoignoreargs_ut/umockautoignoreargs_ut.c [hunk 3]
```diff
@@ -322,4 +322,106 @@ TEST_FUNCTION(umockautoignoreargs_is_call_argument_ignored_for_2nd_arg_when_firs
...
+
+    // act
+    result = umockautoignoreargs_is_call_argument_ignored("WRAPPER(a)(IGNORED_NUM_ARG(", 1, &is_ignored);
+
+    // assert
+    ASSERT_ARE_NOT_EQUAL(int, 0, result);
+}
+
+/* Tests_SRS_UMOCKAUTOIGNOREARGS_01_010: [ umockautoignoreargs_is_call_argument_ignored shall look for the arguments as being the string between the last set of parenthesis in call. ]*/
+TEST_FUNCTION(umockautoignoreargs_is_call_argument_ignored_when_extra_RPAREN_at_end_fails)
```
Comment (mattdurak) [inline tests/umockautoignoreargs_ut/umockautoignoreargs_ut.c]: >RPAREN [](start = 70, length = 6)

LPAREN

File: src/umockautoignoreargs.c [hunk 4]
```diff
@@ -18,164 +20,215 @@ int umockautoignoreargs_is_call_argument_ignored(const char* call, size_t argume
...
+                    if (paren_scope_count == 0)
+                    {
+                        // found the start of the args
+                        break;
+                    }
+                }
+            }
+        } while (i > 0);
+
+        if (is_error)
```
Comment (mattdurak) [inline src/umockautoignoreargs.c]: >is_error [](start = 12, length = 8)

Seems like `paren_scope_count` should be 0 here or there is an error, but the check is missing

File: tests/umockautoignoreargs_ut/umockautoignoreargs_ut.c (hunk 1 above)
Comment (dcristoloveanu) [reply tests/umockautoignoreargs_ut/umockautoignoreargs_ut.c]: Added one test. Note that the compiler will still validate these calls.

---
In reply to: [375595618](https://github.com/Azure/umock-c/pull/140#discussion_r375595618) [](ancestors = 375595618)

File: src/umockautoignoreargs.c (hunk 4 above)
Comment (dcristoloveanu) [reply src/umockautoignoreargs.c]: Added

---
In reply to: [375596015](https://github.com/Azure/umock-c/pull/140#discussion_r375596015) [](ancestors = 375596015)

File: tests/umockautoignoreargs_ut/umockautoignoreargs_ut.c (hunk 3 above)
Comment (dcristoloveanu) [reply tests/umockautoignoreargs_ut/umockautoignoreargs_ut.c]: Fixed.

---
In reply to: [375595720](https://github.com/Azure/umock-c/pull/140#discussion_r375595720) [](ancestors = 375595720)

File: tests/umockautoignoreargs_ut/umockautoignoreargs_ut.c (hunk 2 above)
Comment (dcristoloveanu) [reply tests/umockautoignoreargs_ut/umockautoignoreargs_ut.c]: Fixed.

---
In reply to: [375595686](https://github.com/Azure/umock-c/pull/140#discussion_r375595686) [](ancestors = 375595686)

**Pull request merged**